src/newViews/Tagging/Sources/BulkUpload/Upload/DropboxChooser/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import {
+    Button,
+} from '@the-deep/deep-ui';
+import _ts from '#ts';
+
+import styles from './styles.scss';
+
+export type DropboxLinkType = 'preview' | 'direct';
+
+const pollTime = 3000;
+
+interface Props {
+    className?: string;
+    onSuccess: (files: Dropbox.ChooserFile[]) => void;
+    linkType?: DropboxLinkType;
+    multiselect?: boolean;
+    extensions?: string[];
+    children?: React.ReactNode;
+    disabled?: boolean;
+    folderselect?: boolean;
+    sizeLimit?: number;
+    icons?: React.ReactNode;
+    iconsClassName?: string;
+}
+
+function DropboxChooser(props: Props) {
+    const {
+        className,
+        onSuccess,
+        linkType = 'direct',
+        multiselect,
+        extensions,
+        folderselect,
+        sizeLimit,
+        children,
+        disabled,
+        icons,
+        iconsClassName,
+    } = props;
+
+    const [loaded, setLoaded] = useState(!!window.Dropbox);
+    const [isOpened, setIsOpened] = useState(false);
+
+    useEffect(() => {
+        let timeout: ReturnType<typeof setTimeout>;
+        if (!loaded) {
+            timeout = setTimeout(() => setLoaded(!!window.Dropbox), pollTime);
```

Comment:
Did we mean to use setInterval?

Reply:
@tnagorra I think setTimeout will suffice here